### PR TITLE
Fix codacy issues

### PR DESCRIPTION
**Pull request by casz**
Fixing up Codacy complaints

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by casz** [comment]
Fixed most issues only 8 left :sweat: a few easy once but otherwise some nasty NPath complexity

**Comment by casz** [comment]
This private unused method should be safe to delete?
https://github.com/jenkinsci/configuration-as-code-plugin/blob/dbb5ba265fb5cd6a6289850f8027b31ae78cd2f2/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L633-L635

**Comment by casz** [comment]
@oleg-nenashev this method seems unused? :laughing: 
https://github.com/jenkinsci/configuration-as-code-plugin/blob/dbb5ba265fb5cd6a6289850f8027b31ae78cd2f2/plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java#L51-L53

**Comment by casz** [comment]
Only 5 nasty NPath complexity issues. :sweat:

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java [hunk 1]
```diff
@@ -301,7 +301,7 @@ public CNode describe(T instance, ConfigurationContext context) throws Exception
 
         // add constructor parameters
         for (int i = 0; i < parameters.length; i++) {
-            final Configurator c = context.lookup(attributes[i].getType());
+            context.lookup(attributes[i].getType());
```
Comment (timja) [inline plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java]: is this needed?

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java]: I believe so, due to `context` being used on line 306
```
mapping.put(names[i], attributes[i].describe(instance, context));
```

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java]: Coverage says no :laughing:

File: integrations/pom.xml [hunk 2]
```diff
@@ -408,6 +400,31 @@
         <artifactId>jackson-module-jaxb-annotations</artifactId>
         <version>2.7.5</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
```
Comment (oleg-nenashev) [inline integrations/pom.xml]: Won't really represent the Jenkins behavior. Jenkins core version is used instead in runtime. In JTH this version will be picked instead

File: integrations/pom.xml [hunk 3]
```diff
@@ -408,6 +400,31 @@
...
+        <version>2.2</version>
+      </dependency>
+      <dependency>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>aws-java-sdk</artifactId>
+        <version>1.11.457</version>
+      </dependency>
+      <dependency>
+        <groupId>joda-time</groupId>
+        <artifactId>joda-time</artifactId>
```
Comment (oleg-nenashev) [inline integrations/pom.xml]: Same as for commons-lamg

File: integrations/pom.xml [hunk 4]
```diff
@@ -226,13 +225,6 @@
       <scope>test</scope>
     </dependency>
 
-    <dependency>
-      <groupId>org.jenkins-ci.plugins</groupId>
```
[the remark this answers is not in the document]
Comment (casz) [reply integrations/pom.xml]: Was Duplicated also located here, with higher version https://github.com/jenkinsci/configuration-as-code-plugin/blob/dd42fb7e127012c61aec8ac006304db5ea6907d8/integrations/pom.xml#L110-L115

File: integrations/pom.xml (hunk 2 above)
Comment (casz) [reply integrations/pom.xml]: So what the fix? or does it even matter :question:

File: integrations/pom.xml [hunk 5]
```diff
@@ -408,6 +400,31 @@
...
+      </dependency>
+      <dependency>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>bouncycastle-api</artifactId>
+        <version>2.17</version>
+      </dependency>
+      <dependency>
+        <groupId>org.jenkins-ci.modules</groupId>
+        <artifactId>instance-identity</artifactId>
+        <version>2.2</version>
```
[the remark this answers is not in the document]
Comment (casz) [reply integrations/pom.xml]:
```suggestion
        <version>2.2</version>
        <scope>provided</scope>
```
:question:

[the remark this answers is not in the document]
Comment (oleg-nenashev) [reply integrations/pom.xml]: yes

File: integrations/pom.xml (hunk 2 above)
Comment (oleg-nenashev) [reply integrations/pom.xml]: Pick your poison, one of dependencies declares a version of commons lang while it should not do so. I would rather prefer to add it to the Enforcer ignore list

Comment (casz) [reply integrations/pom.xml]: Shoot, I'll send a PR to plugin-pom 😅
or should we add it here?

Comment (casz) [reply integrations/pom.xml]: I see little difference in excluding vs dependency management.
At least with dependency management, we have control of which version is loaded.

**Comment by casz** [comment]
@oleg-nenashev we could to merge this? :sweat_smile:

**Pull request merged**